server/gitrest/src/routes/git/refs.ts

```diff
@@ -45,16 +49,19 @@ async function getRef(
         // Lookup external storage if commit does not exist.
         const fileName = refId.substring(refId.lastIndexOf("/") + 1);
         // If file does not exist or error trying to look up commit, return the original error.
-        try {
-            const result = await externalStorageManager.read(repo, fileName);
-            if (!result) {
-                winston.error(`getRef error: ${err} repo: ${repo} ref: ${refId}`);
+        if (getRefParams !== undefined && getRefParams.config?.enabled === true)
+        {
+            try {
+                const result = await externalStorageManager.read(repo, fileName);
+                if (!result) {
+                    winston.error(`getRef error: ${err} repo: ${repo} ref: ${refId}`);
+                    return Promise.reject(err);
+                }
+                return getRef(repoManager, owner, repo, refId, getRefParams, externalStorageManager);
+            } catch (bridgeError) {
+                winston.error(`Giving up on creating ref. BridgeError: ${bridgeError}`);
```

Comment:
Other places in FluidFramework/server (such as [Alfred service](https://github.com/microsoft/FluidFramework/blob/main/server/routerlicious/packages/routerlicious-base/src/alfred/app.ts#L105), [Alfred lambda](https://github.com/microsoft/FluidFramework/blob/main/server/routerlicious/packages/lambdas/src/alfred/index.ts#L252), [GitRest](https://github.com/microsoft/FluidFramework/blob/main/server/gitrest/src/externalStorageManager.ts#L41)). Could we keep that pattern here?